rfcs/79-minimal-fragmenting-relationships.md

```diff
@@ -0,0 +1,126 @@
+# Feature Name: `minimal-fragmenting-relationships`
+
+## Summary
+
+The central idea behind fragmenting relationships is the ability to express a connection between two `Entity`s by adding a pair of `(ComponentId, Entity)` as if it were a component. For example if one had several entities representing different kinds of swords one could add the pair `(Equipped, sword_entity)` to the player to signify which is currently equipped. 
+
+Of course you could also just have created an `Equipped` component with an `Entity` field member to represent the same relationship. The advantage to "fragmenting" relationships is each unique pair is treated as if it were a unique component id, forming their own archetypes. This means we could also add `(Equipped, shield_entity)` to the same player and offer performant query APIs that express `(Equipped, *)`to find all equipped pairs.
+
+Another analogy when considering the ECS as an in memory database is that relationships act as foreign keys. This unlocks a potential future where we can express more complex queries matching across multiple entities and relationship pairs. There are a myriad of additional features that can be built on top of the initial implementation described below. The main focus in this RFC will be the blocking issues that must be addressed before the initial implementation is achievable.
+## Motivation
+
+Fragmenting relationships has been a long desired bevy feature. The ability to express and traverse hierarchies and graphs within the ECS unlocks a huge amount of potential. Features such as breadth-first query traversal, immediate hierarchy clean-up, component inheritance, multi-target queries etc. all rely on or are simplified by the existence of fragmenting relationships. 
```

Comment:
```suggestion
Relationships has been a long desired bevy feature. The ability to express and traverse hierarchies and graphs within the ECS unlocks a huge amount of potential. Features such as breadth-first query traversal, immediate hierarchy clean-up, component inheritance, multi-target queries etc. all rely on or are simplified by the existence of some form of relationships in the ECS. 
```
I don't think a specific implementation of relations was the target, just relations in general.

Reply:
@james-j-obrien I think the other possible implementations of relationships have been explored at this point & the nice doogfoody features that comes about from relations are in no small part a result of the implementation being doogfoody. Specifically it makes sense for archetype level capabilities like fast querying, cleanup & traversal to come about from an archetype level implementation in an archetypal ECS.

Reply:
I'm not asking for the RFC to be generalized here, but just to relax the wording for the average user of the feature. This is supposed to be a user facing summary of the motivation, not the exact technical reasons why we're doing this over other implementations.